crates/bevy_core_pipeline/src/skybox/mod.rs

```diff
@@ -70,8 +77,41 @@ impl Plugin for SkyboxPlugin {
 /// To do so, use `EnvironmentMapLight` alongside this component.
 ///
 /// See also <https://en.wikipedia.org/wiki/Skybox_(video_games)>.
-#[derive(Component, ExtractComponent, Clone)]
-pub struct Skybox(pub Handle<Image>);
+#[derive(Component, Clone)]
+pub struct Skybox {
+    pub image: Handle<Image>,
+    /// Scale factor applied to the skybox image.
+    /// After applying this multiplier to the image samples, the resulting values should
+    /// be in units of [cd/m^2](https://en.wikipedia.org/wiki/Candela_per_square_metre).
+    pub brightness: f32,
+}
+
+impl ExtractComponent for Skybox {
+    type Query = (&'static Self, Option<&'static ExposureSettings>);
+    type Filter = ();
+    type Out = (Self, SkyboxUniforms);
+
+    fn extract_component(
+        (skybox, exposure_settings): QueryItem<'_, Self::Query>,
+    ) -> Option<Self::Out> {
+        let exposure = exposure_settings
+            .map(|e| e.exposure())
+            .unwrap_or_else(|| ExposureSettings::default().exposure());
+
+        Some((
+            skybox.clone(),
+            SkyboxUniforms {
+                brightness: skybox.brightness * exposure,
+            },
+        ))
+    }
+}
+
+// TODO: Replace with a push constant once WebGPU gets support for that
```

Comment:
It has support for push constants, but also maybe we don’t want to use push constants for this.

Reply:
No? https://docs.rs/wgpu/latest/wgpu/struct.Features.html#associatedconstant.PUSH_CONSTANTS

Reply:
Hmm… it says it’s native only but I’m pretty sure we’re using push constants emulated by uniforms on WebGL2…

Reply:
https://github.com/bevyengine/bevy/blob/02b520b4e85fa95f1f4a85656a677e3fc2155f36/crates/bevy_pbr/src/render/mesh.rs#L1019

Reply:
Yes, but on WebGPU?

Reply:
You’re surely right. I was just confused because I remembered using them for WebGL2. Resolving.